.github/workflows/retry-worfklow-run.yml

```diff
@@ -0,0 +1,138 @@
+name: Retry workflow run with errors
+
+on:
+  workflow_dispatch:
+    inputs:
+      dry-run:
+        description: |
+          If enabled, only checks for errors in the workflow run and does not retry.
+        default: false
+        type: boolean
+      error-messages:
+        description: |
+          Error messages to look for in annotations and logs.
+          The worfklow run is only retried if any of the error messages are found.
+        required: false
+      run-id:
+        description: Id of the workflow run to retry
+        required: true
+      owner:
+        description: Owner of the GitHub repository of the workflow run to retry
+        required: true
+      repository:
+        description: GitHub repository of the workflow run to retry
+        required: true
+      notify-back-on-error:
+        # Calling workflow must implement the `notify_back_error_message` parameter and forward the error.
+        description: |
+          If enabled, notifies back (i.e. trigger a new run) with an error message when specified error messages are not found.
+        default: false
+        type: boolean
+      notify-back-git-ref:
+        description: Git reference of the calling workflow to notifying back
+        default: main
+      distinct_id:
+        description: |
+          Random distinct Id to easily find the dispatched workflow run
+          Needed by https://github.com/Codex-/return-dispatch as workaround for https://github.com/cli/cli/issues/4001
+        required: false
+
+jobs:
+  retry:
+    name: Retry workflow run
+    runs-on: ubuntu-latest
+    steps:
+      - name: Distinct Id to retrive this worfklow run ${{ inputs.distinct_id }}
+        run: |
+          echo ${{ inputs.distinct_id }}
```

Comment:
non-blocking: 
if the `distinct_id` is unset, this step has an empty output which might be a bit confusing when looking at job logs.

Reply:
I will add a default value then.

Reply:
Done.